### ISSUE TITLE
Why Yolov10 performance on custom dataset worse than yolov7. What hyperparameters can be adjusted to improve performance? 

### ISSUE THREAD

**Issue by mazatov** [ultralytics/ultralytics#15390]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I'm training on my custom object detection dataset. I'm detecting a ball in sports images. So I have one object that is quite small! 

I trained on the same model Yolov10-B and default Yolov7. The performance of Yolov10 was quite a bit worse, which surprised me. My guess is some hyperparameters might need adjusting. One thing I noticed was that Yolov10 exited after 189 epochs because no improvement was observed in 100 epochs while Yolov7 trained for all 300 epochs. 

What hyperparameters can be adjusted to improve performance?

**Comment by Y-T-G** [comment]
Bigger number at the end doesn't necessarily mean that it's a better model all around. It could be better in some aspect and worse in others.

**Comment by mazatov** [comment]
Interesting observation. I'm surprised to hear that, definitely worth a try.

**Comment by pderrenger** [comment]
@mazatov thank you for your feedback. To improve YOLOv10's performance on your custom dataset, consider adjusting hyperparameters such as learning rate, batch size, and image size. Additionally, you might want to experiment with different augmentation techniques and ensure that your dataset is well-balanced. If the issue persists, please verify if it occurs with the latest version of the Ultralytics package.
